Cicd: output screen test on xvfb

liyier90 opened the pull request:
Addresses https://github.com/aisingapore/PeekingDuck/issues/433

- Uses https://github.com/marketplace/actions/gabrielbb-xvfb-action to run `output.screen`tests in a headless environment
  - `output.screen` tests are only run if the platform is Linux because xvfb does not work on other platforms.
- Refactored `output/screen.py` to have 2 constants `MIN_RENDER_SIZE` and `MIN_WINDOW_SIZE` as a workaround to OpenCV quirk
  - The scale factor of 1.5 is arbitrary and can be updated if a larger value is required on other platforms/OSes
- Update GitHub runners to use `ubuntu-latest` and `windows-latest`

ongtw reviewed (changes requested):
Propose changing `output.screen` as follows:
* Current behavior: create `cv2.namedWindow` in `__init__` and use `cv2.resizeWindow` to change window size.
Problem with this is the image is different from the window, so `cv2.getWindowImageRect` returns a different size from the window. Also, there is no way to query the window size; can only `set` it, not `get` it.
* New behavior: Remove `cv2.namedWindow` in `__init__` and resize image directly using `cv2.resize`. When the resized image is displayed using `imshow`, the window will autofit its size.

One issue with this change is the user would not be able to manually resize the window (on Linux at least). But this should not pose a major problem because user can still drag and move window around, which is the more common use case.

liyier90 commented:
- Removed `cv2.namedWindow` and use `cv2.resize()` directly
- Removed any code related to maintaining a min window size
- Removed `pkd_viewer` member variable/config key as it seems to be only used to prevent `cv2.namedWindow` from being called with `--viewer` is used.

Tested on my machine that this also closes https://github.com/aisingapore/PeekingDuck-Private/issues/74

ongtw reviewed (approved):
LGTM
